napari/layers/utils/text.py

```diff
@@ -39,74 +43,68 @@ class TextManager:
-        {'opaque', 'translucent', and 'additive'}. Note that 'opaque` blending
-        is not recommended, as colors the bounding box surrounding the text.
-    visible : bool
-        Set to true of the text should be displayed.
     """
 
-    def __init__(
-        self,
-        text,
-        n_text,
-        properties={},
-        rotation=0,
-        translation=0,
-        anchor='center',
-        color='cyan',
-        size=12,
-        blending='translucent',
-        visible=True,
-    ):
-
-        self.events = EmitterGroup(
-            source=self,
-            auto_connect=True,
-            text=Event,
-            rotation=Event,
-            translation=Event,
-            anchor=Event,
-            color=Event,
-            size=Event,
-            blending=Event,
-            visible=Event,
-        )
-
-        self.events.block_all()
-        self._rotation = rotation
-        self._anchor = Anchor(anchor)
-        self._translation = translation
-        self._color = transform_color(color)[0]
-        self._size = size
-        self._blending = self._check_blending_mode(blending)
-        self._visible = visible
-
-        self._set_text(text, n_text, properties)
-        self.events.unblock_all()
+    text: Union[np.ndarray, str]
+    n_text: int
+    properties: Dict[str, np.ndarray] = field(default_factory=dict)
+    rotation: float = 0.0
+    translation: np.ndarray = np.asarray(0)
+    anchor: Anchor = Anchor.CENTER
+    color: Union[np.ndarray, str] = 'cyan'
+    size: float = 12.0
+    blending: Blending = Blending.TRANSLUCENT
+    visible: bool = True
+
+    def __post_init__(self):
+        # private attributes not in dataclass
+        self._text_format_string: str = ''
+        self._values = None
+        self._mode: TextMode = TextMode.NONE
+        self.anchor = Anchor(self.anchor)
+        self.blending = Blending(self.blending)
+        with self.events.text.blocker():
+            self._set_text(self.text, self.n_text, self.properties)
 
     @property
     def values(self):
         """np.ndarray: the text values to be displayed"""
         return self._values
 
+    @property
+    def mode(self) -> str:
+        """str: The current text setting mode."""
+        return str(self._mode)
+
+    # Things that need to modify the value on set
+
+    def _on_translation_set(self, translation):
+        if not isinstance(translation, np.ndarray):
+            self.translation = np.asarray(translation)
+            return True
+
+    def _on_color_set(self, color):
+        if color != self.color:
+            self.color = transform_color(color)[0]
+        return True
+
+    def _on_blending_set(self, blending):
+        if Blending(blending) == Blending.OPAQUE:
+            warnings.warn(
+                'opaque blending mode is not allowed for text. '
+                'setting to translucent.'
+            )
+            self.blending = Blending.TRANSLUCENT
```

Comment:
here we could just ignore this and leave value unchanged rather that changing to translucent (i know that this matches original behviour, but just noting)

Reply:
I'd be okay with this. I think it would be nice to keep the warning so the user knows that the opaque blending is working (it's just ugly 😛 )

Reply:
I made it the [property setter](https://github.com/napari/napari/blob/32f53357c5d5a9ffce2d57545da9e0f62f680e2c/napari/layers/utils/text.py#L21-L31)